uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-assignment.js

```diff
@@ -342,31 +361,33 @@ define([
                 missingRequiredFields.push('Key date');
               }
 
-              if ($scope.taskList.filter(function(activity){
-                return !activity.activity_type_id;
+              if (enabledTasks.filter(function (task) {
```

Comment:
We are filtering 3 times that means 3 loops, we can optimize this code using a single loop,
```javascript
_.each(enabledTasks, function(task){
 if(!task.activity_type_id && missingRequiredFields.indexOf('Activity type') === -1){ 
   missingRequiredFields.push('Activity type')
 }
 //Same for Assignee and Activity Due Date
})
```